tests/e2e_logging.py

```diff
@@ -1798,6 +1798,168 @@ def test_basic_constraints(network, args):
     assert basic_constraints.value.ca is False
 
 
+def test_etags(network, args):
+    primary, _ = network.find_primary()
+
+    with primary.client("user0") as c:
+        doc = {"id": 999999, "msg": "hello world"}
+        etag = sha256(doc["msg"].encode()).hexdigest()
+
+        # POST ETag matches value
+        r = c.post("/app/log/public", doc)
+        assert r.status_code == http.HTTPStatus.OK
+        assert r.headers["ETag"] == etag, r.headers["ETag"]
```

Comment:
```suggestion

        # POST ETag matches value
        r = c.post("/app/log/public", doc)
        assert r.status_code == http.HTTPStatus.OK
        assert "ETag" in r.headers, r.headers
        etag = r.headers["ETag"]
```

I don't know if should make any assertions about the ETag's value? We happen to know how it's constructed in this instance, but in general its an opaque identifier.

Reply:
I general sure, in this particular test, it seems reasonable to be precise?